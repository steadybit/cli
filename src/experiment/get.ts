// SPDX-License-Identifier: MIT
// SPDX-FileCopyrightText: 2022 Steadybit GmbH

import yaml from 'js-yaml';
import { writeYamlFile } from './files';
import { fetchExperiment } from './api';

export interface Options {
  key: string;
  file?: string;
}

export async function getExperiment(options: Options) {
  const experiment = await fetchExperiment(options.key);

  if (!options.file) {
    console.log(yaml.dump(experiment));
  } else {
    await writeYamlFile(options.file, experiment);
    console.log('Experiment %s written to %s.', options.key, options.file);
  }
}
