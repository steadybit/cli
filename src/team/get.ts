// SPDX-License-Identifier: MIT
// SPDX-FileCopyrightText: 2022 Steadybit GmbH

import { Response } from 'node-fetch';

import { Team, TeamSummary } from './types';
import { executeApiCall } from '../api/http';

export async function getAllTeams(onlyAccessible = true): Promise<Team[]> {
  const response: Response = await executeApiCall({
    method: 'get',
    path: '/api/teams',
    queryParameters: {
      onlyAccessible: String(onlyAccessible)
    }
  });
  const summary: TeamSummary = await response.json();
  return summary.teams;
}
