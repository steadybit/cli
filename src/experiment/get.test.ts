// SPDX-License-Identifier: MIT
// SPDX-FileCopyrightText: 2022 Steadybit GmbH

import { getExperiment } from './get';
import path from 'path';
import * as os from 'os';
import fs from 'fs/promises';

describe('experiment', () => {
  let tmpFolder: string;

  beforeAll(async () => {
    tmpFolder = await fs.mkdtemp(path.join(os.tmpdir(), 'steadybit-cli-test'));
  });

  afterAll(async () => {
    await fs.rm(tmpFolder, { recursive: true });
  });

  describe('get', () => {
    it('should output experiment to file', async () => {
      const file = path.join(tmpFolder, 'experiment.yaml');
      await getExperiment({ key: 'TST-1', file });

      const content = await fs.readFile(file, 'utf-8');
      expect(content).toMatchSnapshot();
    });

    it('should report not found', async () => {
        await expect(getExperiment({ key: 'TST-999' })).rejects.toThrow('Experiment TST-999 not found.');
    });
  });
});
