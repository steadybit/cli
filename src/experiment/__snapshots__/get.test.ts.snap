// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`experiment get should output experiment to file 1`] = `
"key: TST-1
name: Verify TTR fashion bestseller
team: TST
area: Global
environment: Global
lanes:
  - steps:
      - type: action
        ignoreFailure: false
        parameters:
          graceful: 'true'
        actionType: container-stop-attack
        radius:
          targetType: container
          predicate:
            operator: AND
            predicates:
              - key: k8s.namespace
                operator: EQUALS
                values:
                  - steadybit-demo
              - key: k8s.deployment
                operator: EQUALS
                values:
                  - fashion-bestseller
          query: null
          percentage: 50
"
`;
